File: onlinepayments-sdk/src/main/java/com/onlinepayments/sdk/client/android/exception/CommunicationException.java
package com.onlinepayments.sdk.client.android.exception;

import com.onlinepayments.sdk.client.android.model.api.ErrorResponse;

/**
 *
 * Copyright 2020 Global Collect Services B.V
 *
 */
public class CommunicationException extends Exception {

	private static final long serialVersionUID = 378923281056384514L;

	/**
	 * The error response, if this was returned from the server/API
	 */
	public ErrorResponse errorResponse;

	public CommunicationException() {
		super();
	}

	public CommunicationException(String message) {
		super(message);
	}

	public CommunicationException(Throwable t) {
		super(t);
	}

	public CommunicationException(String message, Throwable t) {
		super(message, t);
	}

	public CommunicationException(String message,  ErrorResponse errorResponse) {
		super(message);
		this.errorResponse = errorResponse;
	}

}

File: onlinepayments-sdk/src/main/kotlin/com/onlinepayments/sdk/client/android/exception/CommunicationException.kt
/*
 * Do not remove or alter the notices in this preamble.
 *
 * Copyright © 2025 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */
package com.onlinepayments.sdk.client.android.exception

import com.onlinepayments.sdk.client.android.model.api.ErrorResponse

class CommunicationException(
    message: String? = null,
    cause: Throwable? = null,
    @Suppress("unused") val errorResponse: ErrorResponse? = null,
    @Suppress("unused") val responseBody: String? = null
) : Exception(message, cause) {

    companion object {
        @Suppress("Unused")
        private const val serialVersionUID = 378923281056384514L
    }
}
